x-pack/metricbeat/module/aws/awshealth/awshealth.go

```diff
@@ -0,0 +1,314 @@
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	var config aws.Config
+	if err := m.Module().UnpackConfig(&config); err != nil {
+		return err
+	}
+
+	awsConfig := m.MetricSet.AwsConfig.Copy()
+
+	health_client := health.NewFromConfig(awsConfig, func(o *health.Options) {
+		if config.AWSConfig.FIPSEnabled {
+			o.EndpointOptions.UseFIPSEndpoint = awssdk.FIPSEndpointStateEnabled
+		}
+	})
+	events := m.getEventDetails(ctx, health_client)
+	for _, event := range events {
+		report.Event(event)
+	}
+
+	return nil
+}
+
+// getEventDetails retrieves AWS health events and their details using the provided AWS Health client.
+// It returns a list of Metricbeat events containing relevant AWS health information.
+func (m *MetricSet) getEventDetails(
+	ctx context.Context,
+	awsHealth *health.Client,
+) []mb.Event {
+	// Define event filter to fetch only upcoming and open events
+	eventFilter := types.EventFilter{
+		EventStatusCodes: []types.EventStatusCode{
+			types.EventStatusCodeUpcoming,
+			types.EventStatusCodeOpen,
+		},
+	}
+	var (
+		deEvents          []types.Event
+		affPage           health.DescribeAffectedEntitiesPaginator
+		healthDetails     []AWSHealthMetric
+		healthDetailsTemp []AWSHealthMetric
+		affEntityTemp     AffectedEntityDetails
+		affInputParams    health.DescribeAffectedEntitiesInput
+	)
+
+	// Create an instance of DescribeEventsInput with desired parameters
+	deInputParams := health.DescribeEventsInput{
+		Filter: &eventFilter,
+	}
+
+	// Define options for DescribeEventsPaginator
+	deOptions := &health.DescribeEventAggregatesPaginatorOptions{
+		Limit:                10,
+		StopOnDuplicateToken: true,
+	}
+
+	// Function option to apply options to the paginator
+	deOptFn := func(options *health.DescribeEventsPaginatorOptions) {
+		// Apply the provided options
+		options.Limit = deOptions.Limit
+		options.StopOnDuplicateToken = deOptions.StopOnDuplicateToken
+	}
+	// Define options for DescribeAffectedEntitiesPaginator
+	affOptions := &health.DescribeAffectedEntitiesPaginatorOptions{
+		Limit:                10,
+		StopOnDuplicateToken: true,
+	}
+	// Function option to apply options to the paginator
+	affOptFn := func(options *health.DescribeAffectedEntitiesPaginatorOptions) {
+		// Apply the provided options
+		options.Limit = affOptions.Limit
+		options.StopOnDuplicateToken = affOptions.StopOnDuplicateToken
+	}
+	// Create DescribeEventsPaginator with AWS Health client and options
+	dePage := health.NewDescribeEventsPaginator(awsHealth, &deInputParams, deOptFn)
+
+	for dePage.HasMorePages() {
+		healthDetailsTemp = []AWSHealthMetric{}
+
+		// Perform actions for the current page
+		currentPage, err := dePage.NextPage(ctx)
+		if err != nil {
+			m.Logger().Errorf("[AWS Health] DescribeEvents failed with : %w", err)
+			break
+		}
+		deEvents = currentPage.Events
+		eventArns := make([]string, len(deEvents))
+		// Iterate through events to extract relevant information
+		for i := range deEvents {
```

Comment:
we can simplify the code below by using the value as well as the index in the range expression i.e. `for i, event := range deEvents  { ...`

Reply:
Made changes as suggested.